pkg/discovery/service_test.go

```diff
@@ -28,3 +28,34 @@ func TestCache_Refresh(t *testing.T) {
 		}
 	})
 }
+
+func TestCache_Graveyard(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	planet, err := testplanet.New(t, 1, 8, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Check(planet.Shutdown)
+
+	planet.Start(ctx)
+	satellite := planet.Satellites[0]
+	offline := planet.StorageNodes[0].ID()
+
+	k := satellite.Kademlia.Service
+	k.WaitForBootstrap() // redundant, but leaving here to be clear
+
+	seen := k.Seen()
+	assert.NotNil(t, seen)
+
+	err = satellite.Overlay.Service.Delete(ctx, offline)
+	assert.NoError(t, err)
+
+	time.Sleep(5 * time.Second)
```

Comment:
Note #1275 adds possibility to force a particular system to refresh, so you don't have to sleep anymore.

Reply:
Ah, okay when this was written that hadn't been merged yet. I'll go update this to use the sync2 cycle stuff

Reply:
Updated 👍